aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -317,16 +321,25 @@ std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool kee
 }
 
 std::tuple<Tensor, Tensor> max(const Tensor& self, int64_t dim, bool keepdim) {
-  Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
-  if (self.is_quantized()) {
-    Tensor max = at::empty({0}, self.options().dtype(toUnderlying(self.scalar_type())));
-    at::native::max_out(self.int_repr(), dim, keepdim, max, max_indices);
-    // TODO: qscheme
-    return std::tuple<Tensor, Tensor>(at::_make_per_tensor_quantized_tensor(max, self.q_scale(), self.q_zero_point()), max_indices);
-  } else {
-    Tensor max = at::empty({0}, self.options());
+  if (self.numel() == 0) {
+    Tensor max = at::empty(self.sizes(), self.options());
```

Comment:
Why do you create a large tensor that might be resized later if `keepdim=False`?
Why don't you do as below?
`Tensor max = at::empty({0}, self.options());`
Same in the next line.

Reply:
Works.